sstables/sstables.cc

```diff
@@ -2518,9 +2518,12 @@ static future<bool> do_validate_uncompressed(input_stream<char>& stream, const c
         offset += buf.size();
     }
 
-    if (!stream.eof()) {
-        sstlog.error("Chunk count mismatch between CRC.db and Data.db at offset {}: expected {} chunks but data file has more", offset, checksum.checksums.size());
-        valid = false;
+    {
+        auto buf = co_await stream.read_exactly(4 * 1024);
+        if (!buf.empty() || !stream.eof()) {
```

Comment:
nitpick: no need to check eof() - read_exactly returning an empty buffer already means exactly what you want, that there were no more bytes to read.

By the way, you can also read 1 byte, no need to specifically read 4 * 1024 bytes - readers might think this number has some sort of significance.

Reply:
Oh, and you can also do read() instead of read_exactly(N) with any specific N.